src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnection.cs

```diff
@@ -1864,14 +1873,22 @@ private static void ChangePassword(string connectionString, SqlConnectionString
         // this only happens once per connection
         // SxS: using named file mapping APIs
 
-        internal void RegisterForConnectionCloseNotification<T>(ref Task<T> outerTask, object value, int tag)
+        internal Task<T> RegisterForConnectionCloseNotification<T>(Task<T> outerTask, object value, int tag)
```

Comment:
This could be a candidate for rewriting with simple async/await

Reply:
@David-Engel @cheenamalhotra I can give this a try in the PR if you like or we can leave it for a later date, what do you think?

Reply:
Are there more similar cases that we would like to address?

Reply:
The original code tooks a ref Task<T> and added a continuation to it then returned the new wrapper task in the same variable. It doesn't do any awaiting and if the caller was rewritten to use language async then this function would be totally rewritten or removed as well. It'd prefer not to touch it further in this PR and work on it at a stage when a larger scale transition from contiunation to langauge async is being done.